components/script/dom/webglprogram.rs

```diff
@@ -144,6 +145,9 @@ impl WebGLProgram {
         }
 
         shader_slot.set(None);
+        // TODO: This is not complete because shaders can be attached to multple programs
```

Comment:
I wonder if we could hold a count instead. Increment on attach and decrement on detach.